Src/EB/AMReX_WriteEBSurface.cpp

```diff
@@ -24,25 +24,62 @@ void WriteEBSurface (const BoxArray & ba, const DistributionMapping & dmap, cons
 
         const auto & sfab    = static_cast<EBFArrayBox const &>(mf_ba[mfi]);
         const auto & my_flag = sfab.getEBCellFlagFab();
+        const auto * my_flag_ptr = &my_flag;
 
         const Box & bx = mfi.validbox();
 
         if (my_flag.getType(bx) == FabType::covered ||
             my_flag.getType(bx) == FabType::regular) { continue; }
 
-        std::array<const MultiCutFab *, AMREX_SPACEDIM> areafrac;
-        const MultiCutFab * bndrycent;
-
-        areafrac  =   ebf->getAreaFrac();
-        bndrycent = &(ebf->getBndryCent());
+        std::array<const CutFab *, AMREX_SPACEDIM> areafrac;
+        const CutFab * bndrycent;
+
+        for (int d = 0; d < AMREX_SPACEDIM; ++d) {
+            areafrac[d]  =   &(*ebf->getAreaFrac()[d])[mfi];
+        }
+        bndrycent = &(ebf->getBndryCent()[mfi]);
+
+#ifdef AMREX_USE_GPU
+        std::unique_ptr<EBCellFlagFab> host_flag;
+        if (my_flag.arena()->isManaged() || my_flag.arena()->isDevice()) {
+            host_flag = std::make_unique<EBCellFlagFab>(my_flag.box(), my_flag.nComp(),
+                                                  The_Pinned_Arena());
+            Gpu::dtoh_memcpy_async(host_flag->dataPtr(), my_flag.dataPtr(),
+                                   my_flag.size()*sizeof(Real));
```

Comment:
```suggestion
                                   host_flag.nBytes());
```
It's incorrect to use `sizeof(Real)` because the type is not `Real`.